src/libs/label-detail/components/evidence/index.js

```diff
@@ -41,17 +41,25 @@ const Evidence = ({ collapseId, request, label, type }) => {
       isInt(request.index[0]) &&
       isInt(request.index[1])
     ) {
-      const maxChars = 475
-      const data = { leading: "", middle: "", trailing: "" }
-      data.middle = request.snippet.slice(request.index[0], request.index[1])
-
-      data.leading = request.snippet.slice(0, request.index[0])
-      data.trailing = request.snippet.slice(request.index[1], request.snippet.length)
-
-      data.leading = "... " + data.leading.slice(maxChars * -1)
-      data.trailing = data.trailing.slice(0, maxChars) + " ..."
-
-      return data
+      if (request.cookie) {
+        const data = { leading: "", middle: "", trailing: "" }
+        data.middle = request.snippet.slice(request.index[0], request.index[1])
+        data.leading = request.snippet.slice(0, request.index[0])
+        data.trailing = request.snippet.slice(request.index[1], request.snippet.length)
+        return data
+      } else {
+        const maxChars = 475
+        const data = { leading: "", middle: "", trailing: "" }
+        data.middle = request.snippet.slice(request.index[0], request.index[1])
+
+        data.leading = request.snippet.slice(0, request.index[0])
+        data.trailing = request.snippet.slice(request.index[1], request.snippet.length)
+
+        data.leading = "... " + data.leading.slice(maxChars * -1)
+        data.trailing = data.trailing.slice(0, maxChars) + " ..."
```

Comment:
Unrelated to your PR but is this necessary? We already cut down the snippet in `cutDownSnippet` in `addEvidence.js`.

Reply:
It would matter if we have a user that turns on "save full snippets"